pkg/definition/gen_sdk/gen_sdk.go

```diff
@@ -0,0 +1,473 @@
+		err = g.GenOpenAPISchema(template)
+		if err != nil {
+			if strings.Contains(err.Error(), "unsupported node string (*ast.Ident)") {
+				// https://github.com/cue-lang/cue/issues/2259
+				klog.Warningf("Skip generating OpenAPI schema for %s, known issue: %s", cuePath, err.Error())
+				continue
+			}
+			return errors.Wrapf(err, "generate OpenAPI schema")
+		}
+		err = g.GenerateCode()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetDefinitionValue returns a value.Value from cue bytes
+func (g *Generator) GetDefinitionValue(cueBytes []byte) (*value.Value, error) {
+	g.def = definition.Definition{Unstructured: unstructured.Unstructured{}}
+	if err := g.def.FromCUEString(string(cueBytes), g.meta.config); err != nil {
+		return nil, errors.Wrapf(err, "failed to parse CUE")
+	}
+	g.name = g.def.GetName()
+	g.kind = g.def.GetKind()
+
+	templateString, _, err := unstructured.NestedString(g.def.Object, definition.DefinitionTemplateKeys...)
+	if err != nil {
+		return nil, err
+	}
+	template, err := value.NewValue(templateString+velacue.BaseTemplate, nil, "")
+	if err != nil {
+		return nil, err
+	}
+	return template, nil
+}
+
+// GenOpenAPISchema generates OpenAPI json schema from cue.Instance
+func (g *Generator) GenOpenAPISchema(val *value.Value) error {
+	var err error
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("invalid cue definition to generate open api: %v", r)
+			debug.PrintStack()
+			return
+		}
+	}()
+	if val.CueValue().Err() != nil {
+		return val.CueValue().Err()
+	}
+	paramOnlyVal, err := common.RefineParameterValue(val)
+	if err != nil {
+		return err
+	}
+	defaultConfig := &openapi.Config{ExpandReferences: false, NameFunc: func(val cue.Value, path cue.Path) string {
+		sels := path.Selectors()
+		lastLabel := sels[len(sels)-1].String()
+		return strings.TrimPrefix(lastLabel, "#")
+	}, DescriptionFunc: func(v cue.Value) string {
+		for _, d := range v.Doc() {
+			if strings.HasPrefix(d.Text(), "+usage=") {
+				return strings.TrimPrefix(d.Text(), "+usage=")
+			}
+		}
+		return ""
+	}}
+	b, err := openapi.Gen(paramOnlyVal, defaultConfig)
+	if err != nil {
+		return err
+	}
+	doc, err := openapi3.NewLoader().LoadFromData(b)
+	if err != nil {
+		return err
+	}
+
+	g.completeOpenAPISchema(doc)
+	openapiSchema, err := doc.MarshalJSON()
+	g.openapiSchema = openapiSchema
+	return err
+}
+
+func (g *Generator) completeOpenAPISchema(doc *openapi3.T) {
+	for key, schema := range doc.Components.Schemas {
+		switch key {
+		case "parameter":
+			spec := g.name + "-spec"
+			schema.Value.Title = spec
+			completeFreeFormSchema(schema)
+			completeSchemas(schema.Value.Properties)
+			doc.Components.Schemas[spec] = schema
+			delete(doc.Components.Schemas, key)
+		default:
+			completeSchema(key, schema)
+		}
+	}
+}
+
+// GenerateCode will call openapi-generator to generate code and modify it
+func (g *Generator) GenerateCode() (err error) {
+	tmpFile, err := os.CreateTemp("/tmp", g.name+"-*.json")
```

Comment:
```suggestion
	tmpFile, err := os.CreateTemp("", g.name+"-*.json")
```
Will `/tmp` work in Windows?

Reply:
Haven't validate that. Good catch!